presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -161,6 +169,82 @@ public BaseJdbcClient(JdbcConnectorId connectorId, BaseJdbcConfig config, String
         }
     }
 
+    /**
+     * Looks up the table name given to map it to it's original, case-sensitive name in the database.
+     *
+     * @param connection A valid connection in case the mapping needs to be loaded
+     * @param jdbcSchemaName The database schema name
+     * @param jdbcTableName The table name within the schema that is being used
+     * @return The mapped case-sensitive table name, if found, otherwise the original table name passed in
+     */
+    private String getMappedTableName(Connection connection, String jdbcSchemaName, String jdbcTableName)
+    {
+        Map<String, String> tableMapping = schemaTableMapping.get(jdbcSchemaName);
+
+        // If the mapping doesn't exist or is empty, assume it hasn't been loaded so load it
+        if (tableMapping == null || tableMapping.isEmpty()) {
+            loadTableMap(connection, jdbcSchemaName);
+            tableMapping = schemaTableMapping.get(jdbcSchemaName);
+        }
+        String tmpTableName = jdbcTableName.toLowerCase(ENGLISH);
+        // If this table isn't mapped assume it's a new table, need to reload the mapping
+        if (!tableMapping.containsKey(tmpTableName)) {
+            loadTableMap(connection, jdbcSchemaName);
+        }
+
+        tmpTableName = tableMapping.get(tmpTableName);
+        // If it is NULL at this point, we've hit a small race condition where another thread updated the Map and the table we are looking for was removed
+        // from the database and so no longer has a value or is in the Map. This is unlikely to have happened, but providing NULL protection here just in case.
+        if (tmpTableName != null) {
+            return tmpTableName;
+        }
+        return jdbcTableName;
+    }
+
+    /**
+     * Loads the case-sensitive table mapping for the provided database schema name.
+     *
+     * @param connection The connection to use
+     * @param jdbcSchemaName The schema on that connection to lookup all of the tables for to create the mapping
+     */
+    private void loadTableMap(Connection connection, String jdbcSchemaName)
+    {
+        // Only have 1 thread at a time load the mapping in. This may result in having some queries not return anything or fail because they table
+        ReentrantLock lock = schemaTableMappingLock.get(jdbcSchemaName);
+        if (lock == null) {
```

Comment:
You could use [ConcurrentMap::computeIfAbsent](http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentMap.html#computeIfAbsent-K-java.util.function.Function-) here